src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,210 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Hind Montassif
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from os.path import join, split, exists, expanduser
+from os import makedirs, remove, environ
+
+from urllib.request import urlretrieve
+import hashlib
+import shutil
+
+import numpy as np
+
+def _get_data_home(data_home = None):
+    """
+    Return the path of the remote datasets directory.
+    This folder is used to store remotely fetched datasets.
+    By default the datasets directory is set to a folder named 'gudhi_data' in the user home folder.
+    Alternatively, it can be set by the 'GUDHI_DATA' environment variable.
```

Comment:
The explanation about GUDHI_DATA has mostly disappeared now that this function is private. Maybe we could say a word about it at the beginning of the section in the doc?

Reply:
Tiny nitpick, but: the [`XDG_STATE_HOME`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) environment variable seems relevant here. Something like `XDG_STATE_HOME/gudhi` seems like an appropriate default, rather than introducing a new GUDHI-specific environment variable.

Reply:
Maybe XDG_DATA_HOME (or XDG_CACHE_HOME, or the corresponding default if it isn't defined). But that's rather specific to linux, or the variants of Unix that use freedesktop. On Mac, it could be in `~/Library` (or `~/Library/Caches`) and on windows `%LOCALAPPDATA%` (or `%APPDATA%` or `%TEMP%`), and I don't know where others store their files. If we decide to go that way, it gives a reason to make `get_data_home()` public again so people have a chance to see where the files go. It doesn't necessarily contradict having GUDHI_DATA, those can be fallbacks.
@Hind-M , do you think it is worth the trouble?

Reply:
>  I also find `GUDHI_DATA` and its default location more user-friendly than system folders names...

For a cache, being hidden can be considered a feature :wink: